src/InternPool.zig

```diff
@@ -65,19 +65,49 @@ pub const single_threaded = builtin.single_threaded or !want_multi_threaded;
 pub const TrackedInst = extern struct {
     file: FileIndex,
     inst: Zir.Inst.Index,
-    comptime {
-        // The fields should be tightly packed. See also serialiation logic in `Compilation.saveState`.
-        assert(@sizeOf(@This()) == @sizeOf(FileIndex) + @sizeOf(Zir.Inst.Index));
-    }
+
+    pub const MaybeLost = extern struct {
```

Comment:
Would you mind documenting the role of this type, as well as `TrackedInst`?

I'm struggling to follow along with most of this PR.